spec/module_validator.rb

```diff
@@ -0,0 +1,167 @@
+require 'active_model'
+
+class ArrayInclusionValidator < ActiveModel::EachValidator
```

Comment:
I wish there was a way to namespace this better so it wasn't registered with ActiveRecord directly, as we wouldn't want tests impacting real code

Reply:
Let's introduce a new namespace here of `ModuleValidation`, and we can move this 'generic' array validator to be in that namespace to avoid it leaking into the global namespace.

I believe we can change this to:

```ruby
module ModuleValidation
  class ArrayInclusionValidator < ActiveModel::EachValidator
  end

  # class ReferenceValidator < ArrayInclusionValidator
  # ... call super after doing a references.map { | ref| ref.whatever } ...
  # end

  class Validator
  # ...
# before:    
#         mod.validates :side_effects,
#                  array_inclusion: { in: VALID_SIDE_EFFECT_VALUES }

    mod.validates :side_effects,
                  :'module_validation/array_inclusion' => { in: VALID_SIDE_EFFECT_VALUES }
 
  # ...
  end
end
```

We'll have to rename the file as well `module_validation.rb`; We'll want to update the references to `ModuleValidator` to to `ModuleValidation::Validator`